Config/Parser/AnnotationParser.php

```diff
@@ -0,0 +1,588 @@
+    }
+
+    /**
+     * Return the common field type, like ID, Int, String, and other user-created type.
+     *
+     * @param string $name
+     * @param array  $annotation
+     *
+     * @return array|null
+     */
+    protected static function getGraphQLScalarFieldType($name, $annotation)
+    {
+        // Get the current type, depending on current annotation
+        $type = $graphQLType = null;
+        $nullable = $isMultiple = false;
+        if (array_key_exists('GraphQLColumn', $annotation) && array_key_exists('type', $annotation['GraphQLColumn'])) {
+            $annotation = $annotation['GraphQLColumn'];
+            $type = $annotation['type'];
+        } elseif (array_key_exists('GraphQLToMany', $annotation) && array_key_exists('target', $annotation['GraphQLToMany'])) {
+            $annotation = $annotation['GraphQLToMany'];
+            $type = $annotation['target'];
+            $isMultiple = $nullable = true;
+        } elseif (array_key_exists('GraphQLToOne', $annotation) && array_key_exists('target', $annotation['GraphQLToOne'])) {
+            $annotation = $annotation['GraphQLToOne'];
+            $type = $annotation['target'];
+            $nullable = true;
+        } elseif (array_key_exists('OneToMany', $annotation) && array_key_exists('targetEntity', $annotation['OneToMany'])) {
+            $annotation = $annotation['OneToMany'];
+            $type = $annotation['targetEntity'];
+            $isMultiple = $nullable = true;
+        } elseif (array_key_exists('OneToOne', $annotation) && array_key_exists('targetEntity', $annotation['OneToOne'])) {
+            $annotation = $annotation['OneToOne'];
+            $type = $annotation['targetEntity'];
+            $nullable = true;
+        } elseif (array_key_exists('ManyToMany', $annotation) && array_key_exists('targetEntity', $annotation['ManyToMany'])) {
+            $annotation = $annotation['ManyToMany'];
+            $type = $annotation['targetEntity'];
+            $isMultiple = $nullable = true;
+        } elseif (array_key_exists('ManyToOne', $annotation) && array_key_exists('targetEntity', $annotation['ManyToOne'])) {
+            $annotation = $annotation['ManyToOne'];
+            $type = $annotation['targetEntity'];
+            $nullable = true;
+        } elseif (array_key_exists('Column', $annotation) && array_key_exists('type', $annotation['Column'])) {
+            $annotation = $annotation['Column'];
+            $type = $annotation['type'];
+        }
+
+        if (!$type) {
+            return null;
+        }
+
+        if (array_key_exists('nullable', $annotation)) {
+            $nullable = 'true' == $annotation['nullable']
+                ? true
+                : false;
+        }
+
+        $type = explode('\\', $type);
+        $type = $type[count($type) - 1];
+
+        // Get the graphQL type representation
+        // Specific case for ID and relation
+        if ('id' === $name && 'integer' === $type) {
+            $graphQLType = 'ID';
```

Comment:
I'm probably misunderstanding `ID` field type, but what happens if field `id` is an MySQL auto-incremental integer type? :thinking:

Reply:
It's a bad condition i think. It's just based on my own code, and symfony make this annotation by default:

    /**
     * @ORM\Column(type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */

But yes i think the best case is yo check if the ORM\Id annotation is present :)